lms/djangoapps/course_blocks/transformers/tests/test_library_content.py

```diff
@@ -167,3 +168,159 @@ def test_content_library(self):
+                                                    ]
+                                                },
+                                                {
+                                                    'metadata': {'display_name': "CL Vertical 4"},
+                                                    '#type': 'vertical',
+                                                    '#ref': 'vertical4',
+                                                    '#children': [
+                                                        {
+                                                            'metadata': {'display_name': "HTML3"},
+                                                            '#type': 'html',
+                                                            '#ref': 'html3',
+                                                        }
+                                                    ]
+                                                }
+                                            ]
+                                        }
+                                    ],
+                                }
+                            ],
+                        }
+                    ],
+                }
+            ]
+        }]
+
+    @mock.patch('lms.djangoapps.course_blocks.transformers.library_content.get_student_module_as_dict')
+    def test_content_library_randomize(self, mocked):
+        """
+        Test whether the order of the children blocks matches the order of the selected blocks when
+        course has content library section
+        """
+        mocked.return_value = {
+            'selected': [
+                ['vertical', 'vertical_vertical3'],
+                ['vertical', 'vertical_vertical2'],
+                ['vertical', 'vertical_vertical4'],
+            ]
+        }
+        for i in range(5):
+            trans_block_structure = get_course_blocks(
+                self.user,
+                self.course.location,
+                self.transformers,
+            )
+            children = []
+            for block_key in trans_block_structure.topological_traversal():
+                if block_key.block_type == 'library_content':
+                    children = trans_block_structure.get_children(block_key)
+                    break
+
+            expected_children = ['vertical_vertical3', 'vertical_vertical2', 'vertical_vertical4']
+            self.assertEqual(
+                expected_children,
+                [child.block_id for child in children],
+                u"Expected 'selected' equality failed in iteration {}.".format(i)
+            )
+
+    @mock.patch('lms.djangoapps.course_blocks.transformers.library_content.get_student_module_as_dict')
+    def test_content_library_randomize_selected_blocks_mismatch(self, mocked):
+        # The block structure will contain all the blocks in the course. So the selections
+        # returned from the database will trigger a mismatch
```

Comment:
Could you please move this comment into the docstring and expand the docstring add more explanation of this test? Without the context that you have in the PR description, this test is pretty confusing.